turbopack/crates/turbo-tasks-fs/src/rope.rs

```diff
@@ -148,12 +151,41 @@ impl From<Cow<'static, [u8]>> for Rope {
         } else {
             Rope {
                 length: bytes.len(),
-                data: InnerRope(Arc::new(vec![Local(bytes)])),
+                data: InnerRope(Arc::new(vec![compress_bytes(bytes)])),
             }
         }
     }
 }
 
+fn compress_bytes(bytes: Cow<'static, [u8]>) -> RopeElem {
+    debug_assert!(!bytes.is_empty(), "must not have empty bytes");
+
+    match bytes {
+        Cow::Borrowed(b) => Static(b),
+        Cow::Owned(b) => {
+            let mut encoder = lz4::EncoderBuilder::new()
```

Comment:
should we add a size threshold here?

`if b.len() < 256` or something like that?   or i guess we could simply check that `output.len() < b.len()` and only store the compressed bytes if they are actually smaller.

For our expected inputs i suspect that compression is always a win, but presumably some very small byte slices are sometimes appended to a rope

Reply:
IRL we discussed this a bit.

1. We could avoid a new enum arm of `Compressed` by using a heuristic based on `length`, however this is subject to some subtle special cases (e.g. compression could produce an alternate text that is the same size!), we could workaround this of course, but it means places reading need to check...so using an enum variant seems better safer.
2. It is unclear to us whether or not RopeBuilder does a good job of producing large compressible chunks.  So RopeBuilder might need some new heuristics to make this worthwhile.

Reply:
If this is an issue, lz4 can do a much better on small inputs if provided with a pre-trained dictionary of representative data (https://docs.rs/lzzzz/2.0.0/lzzzz/?search=dict)